pytools/tag.py

```diff
@@ -324,6 +325,15 @@ def tags_of_type(self, tag_t: Type[TagT]) -> FrozenSet[TagT]:
 # }}}
 
 
+# {{{ IgnoredForEqualityTag
+
+class IgnoredForEqualityTag(Tag):
```

Comment:
Please override `Taggable.__eq__` and include a test.

Reply:
I'm not 100% sure what you had in mind, but how does [f921af0](https://github.com/inducer/pytools/pull/134/commits/f921af0e5be2055691132ea030d327b5e99caebe) look?

Reply:
`__hash__` must also be overridden, I think.

Reply:
What do you think of [4320ef3](https://github.com/inducer/pytools/pull/134/commits/4320ef352623b2cbe16917c5c6033a4dc5925f92) ?

Reply:
That looks good, thanks!

Reply:
That's not quite what I had in mind (I figured you'd override `Taggable.__eq__`), but it's an interesting approach. Unfortunately, I think there are two aspects that make it impractical:

- We now can't have more than one of these in the set of tags (they all compare equal) after all.
- They'll still prompt inequality between objects that do and do not have them.

Reply:
The problem with the Taggable approach would be every downstream user would have to call `super().__eq__`, I think.

Reply:
I don't see how we can resolve these issues here. Should we just use the localized approach in pytato (https://github.com/inducer/pytato/pull/284/commits/0794bdb4fd986549bce4ed6f463e31f34f995dec) @kaushikcfd @inducer ?

Reply:
> The problem with the Taggable approach would be every downstream user would have to call `Taggable.__eq__`, I think.

Yes, IMO that's expected, and a reasonable thing to do anyhow.

Reply:
FWIW, we would need to investigate how dataclasses handle this.

Reply:
As you can tell, I'm not a fan of the localized approach. We can either do this once, and consistently, or we'll keep hackering around with this nonsense forever.